### PR TITLE
[CDAP-5662] -Hydrator++ bug fix 5

### PR DESCRIPTION
**Pull request by ajainarayanan**
- CDAP-5662 : If no system artifacts the UI should show appropriate error messages.
- Fixes side panel to show no artifacts/plugins found message when no artifacts or plugins are present.

### CODE REVIEW

File: cdap-ui/app/features/hydratorplusplus/routes.js [hunk 1]
```diff
@@ -57,42 +57,88 @@ angular.module(PKG.name + '.feature.hydratorplusplus')
...
-              }).$promise.then((res) => {
-                let uiSupportedArtifacts = ['cdap-etl-batch', 'cdap-etl-realtime', 'cdap-data-pipeline'];
-                let filteredRes;
-                if ($stateParams.artifactType) {
-                  filteredRes = res.filter( r => r.name === $stateParams.artifactType );
-                  if (filteredRes.length) {
-                    defer.resolve(filteredRes[0].name);
+              }).$promise.then((artifactsFromBackend) => {
+                let showWarningAndNavigateAway = () => {
+                  if (!$state.current.name.length) {
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/routes.js]: Is this scenario when user directly goes to the URL?

File: cdap-ui/app/services/constants.js [hunk 2]
```diff
@@ -85,7 +85,8 @@ angular.module(PKG.name + '.services')
               'INVALID-SCHEDULE': 'Batch pipeline should have a valid schedule specification.',
               'INVALID-INSTANCE': 'Realtime pipeline should have a valid instance specification.',
               'INVALID-NODES-CONNECTIONS': 'Unknown node(s) in \'connections\' property in pipeline specification.'
-            }
+            },
+            'MISSING-SYSTEM-ARTIFACTS': 'Missing system artifacts. Please load system artifacts to use hydrator studio.'
```
Comment (elfenheart) [inline cdap-ui/app/services/constants.js]: We might need to be more descriptive for this. The user don't necessarily know what system artifacts are missing.

File: cdap-ui/app/features/hydratorplusplus/routes.js (hunk 1 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/hydratorplusplus/routes.js]: Yes.

File: cdap-ui/app/services/constants.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/services/constants.js]: We can change the message to whatever we want. Just externalized it for the same. 👍

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**